ros/api/v1/hosts.py

```diff
@@ -376,57 +380,142 @@ def get(self, host_id):
+        oversized_systems = count_per_state(system_queryset, {'state': "Oversized"})
+        awaiting_data_systems = count_per_state(system_queryset, {'state': "Waiting for data"})
+        idling_systems = count_per_state(system_queryset, {'state': "Idling"})
+
+        non_optimized = system_queryset.filter(
+            or_(
+                System.cpu_state is not None,
+                System.io_state is not None,
+                System.memory_state is not None,
+            ),
+            and_(
+                System.state != 'Optimized',
+                System.state != 'Waiting for data'
+            )
+        )
+        non_optimized_count = non_optimized.count()
+
+        # cpu counts
+        total_cpu_issues = sum([
+            len(system.cpu_state) if system.cpu_state else 0
+            for system in non_optimized.filter(System.cpu_state is not None)
+        ])
+        cpu_undersized = non_optimized.filter(System.cpu_state.any(SubStates.CPU_UNDERSIZED.value)).count()
+        cpu_under_pressure = non_optimized.filter(System.cpu_state.any(
+            SubStates.CPU_UNDER_PRESSURE.value)
+        ).count()
+        cpu_oversized = non_optimized.filter(System.cpu_state.any(SubStates.CPU_OVERSIZED.value)).count()
+
+        # memory counts
+        total_memory_issues = sum([
+            len(system.memory_state) if system.memory_state else 0
+            for system in non_optimized.filter(System.memory_state is not None)
+        ])
+        memory_undersized = non_optimized.filter(System.memory_state.any(SubStates.MEMORY_UNDERSIZED.value)).count()
+        memory_pressure = non_optimized.filter(System.memory_state.any(
+            SubStates.MEMORY_UNDER_PRESSURE.value)
+        ).count()
+        memory_oversized = non_optimized.filter(System.memory_state.any(SubStates.MEMORY_OVERSIZED.value)).count()
+
+        # io counts
+        total_io_issues = sum([
+            len(system.io_state) if system.io_state else 0
+            for system in non_optimized.filter(System.io_state is not None)
+        ])
+        io_undersized = non_optimized.filter(System.io_state.any(SubStates.IO_UNDERSIZED.value)).count()
+        io_pressure = non_optimized.filter(System.io_state.any(SubStates.IO_UNDER_PRESSURE.value)).count()
+        io_oversized = non_optimized.filter(System.io_state.any(SubStates.IO_OVERSIZED.value)).count()
+
+        all_conditions_count = total_cpu_issues + total_memory_issues + total_io_issues
+
         response = {
             "systems_per_state": {
                 "under_pressure": {
-                    "count": 0,
-                    "percentage": 5.57
+                    "count": under_pressure_systems,
+                    "percentage": calculate_percentage(under_pressure_systems, all_systems_count)
                 },
                 "undersized": {
-                    "count": 1,
-                    "percentage": 7.68
+                    "count": undersized_systems,
```

Comment:
@saltgen once Stefan provides the inputs on the final order in the [jira card](https://issues.redhat.com/browse/RHIROS-516), can you follow the same and send response in the same order for state and conditions please?

Reply:
@PreetiW I have pushed the change as per discussion, let me know if changes are required yet again.